libs/tendermint/p2p/conn/connection.go

```diff
@@ -1284,7 +1289,15 @@ func unmarshalPacketFromAminoReader(r io.Reader, maxSize int64) (packet Packet,
 		packet = packetPong
 		return
 	} else if bytes.Equal(PacketMsgTypePrefix, bz[0:4]) {
-		msg := PacketMsg{}
+		var msg *PacketMsg
+		if targetMsg != nil {
+			msg = targetMsg
+			msg.ChannelID = 0
+			msg.EOF = 0
+			msg.Bytes = msg.Bytes[:0]
```

Comment:
maybe could encapsulate a function for the reset？

Reply:
I added Reset method to PacketMsg